util/testutil/integration/containerd.go

```diff
@@ -130,37 +126,20 @@ disabled_plugins = ["cri"]
 		"--containerd-worker-labels=org.mobyproject.buildkit.worker.sandbox=true", // Include use of --containerd-worker-labels to trigger https://github.com/moby/buildkit/pull/603
 	}
 
-	var upt []ConfigUpdater
-
-	for _, v := range conf.mv.values {
-		if u, ok := v.value.(ConfigUpdater); ok {
-			upt = append(upt, u)
-		}
-	}
-
-	if conf.mirror != "" {
-		upt = append(upt, withMirrorConfig(conf.mirror))
-	}
-
-	if len(upt) > 0 {
-		dir, err := writeConfig(upt)
-		if err != nil {
-			return nil, nil, err
-		}
-		deferF.append(func() error {
-			return os.RemoveAll(dir)
-		})
-		buildkitdArgs = append(buildkitdArgs, "--config="+filepath.Join(dir, "buildkitd.toml"))
-	}
-
-	buildkitdSock, stop, err := runBuildkitd(buildkitdArgs, logs, 0, 0)
+	buildkitdSock, stop, err := runBuildkitd(cfg, buildkitdArgs, logs, 0, 0)
 	if err != nil {
+		deferF.F()()
 		printLogs(logs, log.Println)
 		return nil, nil, err
 	}
 	deferF.append(stop)
 
-	return &cdsandbox{address: address, sandbox: sandbox{mv: conf.mv, address: buildkitdSock, logs: logs, cleanup: deferF, rootless: false}}, cl, nil
+	return cdbackend{
```

Comment:
It needed both the containerd worker address and the buildkitd socket, since nothing else would need the containerd worker I didn't want to throw it in the regular backend implementation. That being said, the regular backend implementation is internal so it could just tag along there too if you'd prefer.